fabric-item-api-v1/src/main/java/net/fabricmc/fabric/api/client/item/v1/ItemTooltipDataCallback.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2016, 2017, 2018, 2019 FabricMC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.fabricmc.fabric.api.client.item.v1;
+
+import java.util.List;
+
+import net.minecraft.client.item.TooltipData;
+import net.minecraft.item.ItemStack;
+
+import net.fabricmc.fabric.api.event.Event;
+import net.fabricmc.fabric.api.event.EventFactory;
+
+public interface ItemTooltipDataCallback {
```

Comment:
I would suggest combining this with the `ItemTooltipCallback` class. So you would have one class with two event fields and two functional interfaces. `ServerMessageEvents` in `fabric-message-api` is an example.

Also, you need documentation about this event, like an explanation of what this does. Also, you should explain the difference compared to `ItemTooltipCallback` and what cases this api should be used for.

Reply:
I think the way this PR does it is fine. The XxxCallback format is fine when the event is standalone.

Reply:
I agree with documentation comment: the interface should have a comment explaining the high-level point of the event (even though it's obvious for us, it might not be for beginners), and any useful implementation details such as when it's fired.

Reply:
Important things to document:
- the normal tooltip data is already in the list by the time this event is called.
- all the tooltip components in the list will be rendered, in order.